packages/spotlight/src/spotlight.js

```diff
@@ -377,13 +377,22 @@ const Spotlight = (function () {
 		// trying to focus something else (potentially) unless the window was previously blurred
 		if (_spotOnWindowFocus) {
 			setPlatformPointerMode();
+
 			// If the window was previously blurred while in pointer mode, the last active containerId may
 			// not have yet set focus to its spottable elements. For this reason we can't rely on setting focus
 			// to the last focused element of the last active containerId, so we use rootContainerId instead
-			if (!Spotlight.focus(getContainerLastFocusedElement(rootContainerId))) {
+			let lastFocusedElement = getContainerLastFocusedElement(rootContainerId);
+			while (isContainer(lastFocusedElement)) {
+				({lastFocusedElement} = getContainerConfig(lastFocusedElement));
+			}
+
+			if (!Spotlight.focus(lastFocusedElement)) {
 				// If the last focused element was previously also disabled (or no longer exists), we
-				// need to set focus somewhere
-				Spotlight.focus();
+				// need to set focus to the next target element within the same container
+				if (!Spotlight.focus(getTargetByContainer(last(getContainersForNode(lastFocusedElement))))) {
```

Comment:
I think `getTargetByContainer` here is redundant as `Spotlight.focus()` will resolve that on its own.